File: api/Enkl.Api/Controllers/TemplatesController.cs
using Enkl.Api.Auth;
using Enkl.Api.Dtos;
using Enkl.Api.Services;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;

namespace Enkl.Api.Controllers;

/// <summary>
/// Separate from OrganisationsController since its gating is split, not uniformly OrgAdmin: any
/// signed-in org member may list/read/create a template (same trust level as creating a column or
/// task type on a project they belong to), but renaming/deleting a shared org asset requires OrgAdmin
/// (see TemplateService's own note).
/// </summary>
[ApiController]
[Authorize]
[Route("api/organisations/me/templates")]
public class TemplatesController : ControllerBase
{
    private readonly TemplateService _templates;

    public TemplatesController(TemplateService templates)
    {
        _templates = templates;
    }

    [HttpGet]
    public async Task<IActionResult> List()
    {
        return Ok(await _templates.ListAsync(User.OrgId()));
    }

    [HttpGet("{id:guid}")]
    public async Task<IActionResult> GetDetail(Guid id)
    {
        var result = await _templates.GetDetailAsync(User.OrgId(), id);
        return result is null ? NotFound() : Ok(result);
    }

    [HttpPost]
    public async Task<IActionResult> Create(CreateTemplateRequest request)
    {
        return Ok(await _templates.CreateAsync(User.OrgId(), request));
    }

    [HttpPut("{id:guid}")]
    [Authorize(Policy = "OrgAdmin")]
    public async Task<IActionResult> Rename(Guid id, UpdateTemplateRequest request)
    {
        var ok = await _templates.RenameAsync(User.OrgId(), id, request.Name);
        return ok ? NoContent() : NotFound();
    }

    [HttpDelete("{id:guid}")]
    [Authorize(Policy = "OrgAdmin")]
    public async Task<IActionResult> Delete(Guid id)
    {
        var ok = await _templates.DeleteAsync(User.OrgId(), id);
        return ok ? NoContent() : NotFound();
    }
}

File: mariadb-api/src/Controllers/TemplatesController.php
<?php

declare(strict_types=1);

namespace Enkl\Api\Controllers;

use Enkl\Api\Db\Database;
use Enkl\Api\Services\TemplateService;
use Psr\Http\Message\ResponseInterface as Response;
use Psr\Http\Message\ServerRequestInterface as Request;

/**
 * Ported from Controllers/TemplatesController.cs. Separate from OrganisationsController since its
 * gating is split, not uniformly OrgAdmin — see routes.php, which puts list/detail/create under
 * RequireAuthMiddleware only, and rename/delete under an additional OrgAdminMiddleware.
 */
final class TemplatesController extends BaseController
{
    private function service(): TemplateService
    {
        return new TemplateService(Database::connection());
    }

    public function list(Request $request, Response $response): Response
    {
        return $this->json($response, $this->service()->list($this->callerOrgId($request)));
    }

    public function detail(Request $request, Response $response, array $args): Response
    {
        $result = $this->service()->getDetail($this->callerOrgId($request), $args['id']);
        return $result === null ? $this->notFound($response) : $this->json($response, $result);
    }

    public function create(Request $request, Response $response): Response
    {
        return $this->json($response, $this->service()->create($this->callerOrgId($request), $this->body($request)));
    }

    public function rename(Request $request, Response $response, array $args): Response
    {
        $body = $this->body($request);
        $ok = $this->service()->rename($this->callerOrgId($request), $args['id'], (string) ($body['name'] ?? ''));
        return $ok ? $this->noContent($response) : $this->notFound($response);
    }

    public function delete(Request $request, Response $response, array $args): Response
    {
        return $this->service()->delete($this->callerOrgId($request), $args['id']) ? $this->noContent($response) : $this->notFound($response);
    }
}
